src/Build/Logging/FancyLogger/FancyLoggerNode.cs

```diff
@@ -4,148 +4,136 @@
 
 using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
 using Microsoft.Build.Framework;
 
 namespace Microsoft.Build.Logging.FancyLogger
 { 
-    // TODO: Maybe remove
-    public enum FancyLoggerNodeType
+    public class FancyLoggerProjectNode
     {
-        None,
-        Build,
-        Project,
-        Target,
-        Task,
-        Message,
-        Warning,
-        Error
-    }
+        private static string GetUnambiguousPath(string path)
+        {
+            return Path.GetFileName(path);
+        }
 
-    public class FancyLoggerNode
-    {
-        public string Id;
-        public Dictionary<string, FancyLoggerNode> Children = new Dictionary<string, FancyLoggerNode>();
-        public FancyLoggerNode? Parent;
+        public int Id;
+        public string ProjectPath;
+        public bool Finished;
+        // Line to display project info
         public FancyLoggerBufferLine? Line;
-        public int Depth = 0;
-        public FancyLoggerNode(string id)
+        // Targets
+        public int FinishedTargets;
+        public FancyLoggerBufferLine? CurrentTargetLine;
+        public FancyLoggerTargetNode? CurrentTargetNode;
+        // Messages, errors and warnings
+        List<Object> AdditionalDetails = new();
+        public FancyLoggerProjectNode(ProjectStartedEventArgs args)
         {
-            Id = id;
+            Id = args.ProjectId;
+            ProjectPath = args.ProjectFile!;
+            Finished = false;
+            FinishedTargets = 0;
         }
-        public FancyLoggerNode(int id, FancyLoggerNodeType type)
+
+        public void Log()
         {
-            switch (type)
+            string lineContents = ANSIBuilder.Alignment.SpaceBetween(
+                $"{(Finished ? ANSIBuilder.Formatting.Color("✓", ANSIBuilder.Formatting.ForegroundColor.Green) : ANSIBuilder.Graphics.Spinner())} {ANSIBuilder.Formatting.Dim("Project: ")} {ANSIBuilder.Formatting.Color(ANSIBuilder.Formatting.Bold(GetUnambiguousPath(ProjectPath)), Finished ? ANSIBuilder.Formatting.ForegroundColor.Green : ANSIBuilder.Formatting.ForegroundColor.Default )}",
+                $"({FinishedTargets} targets completed)",
+                Console.WindowWidth
+            );
+            // Create or update line
+            if (Line == null) Line = FancyLoggerBuffer.WriteNewLine(lineContents);
+            else FancyLoggerBuffer.UpdateLine(Line.Id, lineContents);
+            // If current target
+            if (CurrentTargetNode == null) return;
```

Comment:
Understanding check:
This should only return early if you haven't even started building your first target yet, correct?

Reply:
Yes. That is, on project started. On all other cases, it should continue and display the relevant current target and info. 
Will replace the comment so it makes more sense.